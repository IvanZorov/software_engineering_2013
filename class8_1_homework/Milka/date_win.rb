require 'csv'

den=[]
CSV.foreach(ARGV[0]) do |row|
	if ARGV[1]==row[0]
		den << [row[1],row[2]]
	end
end

den.each do |element|
	if element[1].split("-")[0].to_i>element[1].split("-")[1].to_i
		element[0]=element[0].split("-")[0]
		element[1]=element[1].split("-")[0].to_i
	else
		element[0]=element[0].split("-")[1]
		element[1]=element[1].split("-")[1].to_i
	end
end

den.sort!{|a,b| b[1] <=> a[1]}

CSV.open("date_win_result.csv","w") do |csv|
	den.each do |element|
		csv << element
	end
end
