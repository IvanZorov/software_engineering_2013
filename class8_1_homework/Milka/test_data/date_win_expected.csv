LAC,145
BOS,133
CHI,98
